Embedded Workflows on OCP docker custom pull secret (#1762)
ManageIQ/manageiq-documentation: managing_providers/_topics/embedded_workflows.md

You can create and use embedded workflows as needed to not only change parts of the provisioning process, but also to automate other operational tasks.

> Workflows must be authored in Amazon State Languages (ASL) format. As part of authoring a workflow, you (or your users) can build container images that are able to perform any tasks that are required in any language that you like. You can use these images during Task states in your workflows. 

Reviewer: Please remove trailing whitespace

2. Build the docker containers that are required for the workflow.

   When you have the code for your task resource written, you need to bundle it into a docker image. You can bundle the code by creating a standard [Dockerfile](https://docs.docker.com/engine/reference/builder/) and building the image (https://docs.docker.com/engine/reference/commandline/build/). Then, you can push the image to a [registry](https://docs.docker.com/engine/reference/commandline/push/), which makes the image available to be used by {{ site.data.product.title_short }}. When you have pushed your images to an image registry, you can add the registry to {{ site.data.product.title_short }}. 
   
  Pull secrets for containers are used differently between appliances and the OpenShift Container Platform (OCP). These differences are outlined in the following sections.

  #### Running an Embedded Workflow on Appliances

>    * On appliances, `podman` is used to execute the container
> 
>      On appliances, `podman` is used to execute the container so use [podman login](https://docs.podman.io/en/stable/markdown/podman-login.1.html) as the `manageiq` user. 

Reviewer: ```suggestion
   * On appliances, `podman` is used to execute the container so use [podman login](https://docs.podman.io/en/stable/markdown/podman-login.1.html) as the `manageiq` user. 
```

   When you have the code for your task resource written, you need to bundle it into a docker image. You can bundle the code by creating a standard [Dockerfile](https://docs.docker.com/engine/reference/builder/) and building the image (https://docs.docker.com/engine/reference/commandline/build/). Then, you can push the image to a [registry](https://docs.docker.com/engine/reference/commandline/push/), which makes the image available to be used by {{ site.data.product.title_short }}. When you have pushed your images to an image registry, you can add the registry to {{ site.data.product.title_short }}. 
   
  Pull secrets for containers are used differently between appliances and the OpenShift Container Platform (OCP). These differences are outlined in the following sections.

>    * On appliances, `podman` is used to execute the container

Reviewer: This line seems to be duplicated on the next line down

      
      You are recommended to use a docker.io [access token](https://docs.docker.com/security/for-developers/access-tokens/) so that the token does not expire.

>    * Provide an image pull secret to a podified Kubernetes container, and then add it to a service account

Reviewer: Is this section needed now that we have `#### Running an Embedded Workflow on the Openshift Container Platform (OCP) ` below?

     In order to pull an image from a private registry you have to provide an `ImagePullSecret` to your containers, see [Pull an Image from a Private Registry](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/). {{ site.data.product.title_short }} uses a service account called {{ site.data.product.workflow_service_account }} to run containers for your workflows. You can add an `ImagePullSecret` to this service account by following [Add Image Pull Secrets to a service account](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#add-imagepullsecrets-to-a-service-account/).

> #### Running an Embedded Workflow on the Openshift Container Platform (OCP) 

Reviewer: Should there be a `#### Running an Embedded Workflow an Appliances` section above to match this?

  }
}
```

> #### Running an Embedded Workflow on the Openshift Container Platform (OCP) 

Reviewer 2: Very minor, but probably good to add a blank line above this to separate it from the preview section.

```suggestion

#### Running an Embedded Workflow on the Openshift Container Platform (OCP) 
```

Reviewer 2: Also OpenShift should have a capital S

```suggestion
#### Running an Embedded Workflow on the OpenShift Container Platform (OCP) 
```

If the user is running an embedded workflow on OCP, as opposed to appliances, and is using a docker repository requiring a custom pull secret for their container images, the following steps are necessary:

> 1. create a docker pull secret for the repository in question in the same namespace as IM, for example: 

Reviewer 2: "IM" is a downstream specific name, so we should use a conref here.  Also, minor, but this should probably be a full sentence.

```suggestion
1. Create a docker pull secret for the repository in question in the same namespace as {{ site.data.product.title_short }}, for example:
```

Reviewer 2: It says "docker pull secret", but that doesn't sound right to me - I think it should just be "Create a pull secret".  Maybe I'm wrong though - @agrare Thoughts?

Reviewer: Technically you are creating a k8s `Secret` object and the purpose is up to the user, but I think `Pull Secret` or `Image Pull Secret` would be an accurate description.

    ```
    oc create secret docker-registry <pull-secret-name> --docker-server=<docker-server> --docker-username=<docker-username> --docker-password=<docker-password> 
    ```

> 2. add this pull secret to the ibm-infra-management-application-default service account, for example:
> 
>     ```
>     oc secret link ibm-infra-management-application-default <pull-secret-name> --for=pull
>     ```

Reviewer 2: This should also use conrefs, and also be sentence cased.

```suggestion
2. Add this pull secret to the `{{ site.data.product.workflow_service_account }}` service account, for example:

    ```
    oc secret link {{ site.data.product.workflow_service_account }} <pull-secret-name> --for=pull
    ```
```

  }
}
```

> 

Reviewer: Was this removal intentional?  It doesn't seem related to the rest of these changes
